src/core/fifo_pool.mli

```diff
@@ -48,13 +48,17 @@ val with_ : (unit -> (t -> 'a) -> 'a, _) create_args
 (**/**)
 
 module Private_ : sig
-  type state
+  type worker_state
```

Comment:
I feel like this should be an internal detail that's not accessible at all, or only via a more general type, e.g. a function that construct a generic `Moonpool.state`.

Reply:
it's in `Private_`, use at your own risks. It's used in `Moonpool_fiber.main`.